kms/keysource.go

```diff
@@ -92,17 +96,26 @@ func (key *MasterKey) Decrypt() ([]byte, error) {
 		log.WithField("arn", key.Arn).Info("Decryption failed")
 		return nil, fmt.Errorf("Error base64-decoding encrypted data key: %s", err)
 	}
+
+	// Capture service locally to prevent data races
+	kmsSvcMtx.RLock()
+	svc := kmsSvc
```

Comment:
These changes were necessary to deal with the concurrent access of the `kmsSvc` by the decrypt calls.